src/redux/sagas/motions/createEditDomainMotion.ts

```diff
@@ -35,10 +33,175 @@ import {
   uploadAnnotation,
   initiateTransaction,
   createActionMetadataInDB,
+  getPermissionProofsLocal,
+  getChildIndexLocal,
 } from '../utils/index.ts';
 
 // import { ipfsUpload } from '../ipfs';
 
+type CreateTransactionPayloadBase = {
+  isMultiSig: boolean;
+  colonyAddress: Address;
+  batchKey: string;
+  metaId: string;
+  index: number;
+  colonyDomains: Domain[];
+  isCreateDomain: boolean;
+  domainId: number;
+  action: string;
+  colonyRoles: ColonyRoleFragment[];
+};
+
+type CreateTransactionPayloadMotion = {
+  motionDomainId: number;
+};
+
+type CreateTransactionPayload = {
+  base: CreateTransactionPayloadBase;
+  motionPayload: CreateTransactionPayloadMotion;
+};
+
+function* getCreateTransactionPayload(payload: CreateTransactionPayload) {
+  const {
+    base: {
+      isMultiSig,
+      colonyAddress,
+      batchKey,
+      metaId,
+      index,
+      colonyDomains,
+      isCreateDomain,
+      domainId,
+      action,
+      colonyRoles,
+    },
+  } = payload;
+
+  const transactionParams = {
+    context: isMultiSig
+      ? ClientType.MultisigPermissionsClient
+      : ClientType.VotingReputationClient,
+    methodName: TRANSACTION_METHODS.CreateMotion,
+    identifier: colonyAddress,
+    group: {
+      key: batchKey,
+      id: metaId,
+      index,
+    },
+    params: [] as MethodParams,
+    ready: false,
+  };
+
+  const colonyManager = yield getColonyManager();
+
+  const colonyClient = yield colonyManager.getClient(
+    ClientType.ColonyClient,
+    colonyAddress,
+  );
+
+  // eslint-disable-next-line no-warning-comments
+  // TODO: Use Sam's upcoming list of required permissions as the source of truth
+  // src/constants/actions.ts: PERMISSIONS_NEEDED_FOR_ACTION
+  const requiredRoles = [ColonyRole.Architecture];
+
+  const userAddress = yield colonyClient.signer.getAddress();
+
+  if (isMultiSig) {
+    const [permissionDomainId, childSkillIndex] = yield call(
+      getPermissionProofsLocal,
+      {
+        networkClient: colonyClient.networkClient,
+        colonyRoles,
+        colonyDomains,
+        requiredDomainId: Id.RootDomain,
+        requiredColonyRole: requiredRoles,
+        permissionAddress: userAddress,
+        isMultiSig,
+      },
+    );
+
+    const encodedAction = colonyClient.interface.encodeFunctionData(
+      action,
+      [permissionDomainId, childSkillIndex, domainId, '.'], // domainMetadataIpfsHash
+    );
+
+    transactionParams.params = [
+      Id.RootDomain,
+      childSkillIndex,
+      [isCreateDomain ? colonyAddress : AddressZero],
```

Comment:
I don't think this ternary is needed. I tested it with just `AddressZero` and had no problems.

Reply:
Oh did it work when you created a domain? I was having issues with `createDomain` when AddressZero was being used but all right, I shall try again!

Reply:
@iamsamgibbs can we jump on a call regarding this? I'm not able to create a team via multi-sig if I just used AddressZero for some reason:

https://github.com/JoinColony/colonyCDapp/assets/50642296/2afb35cd-906c-4aa3-a9a3-1b05abed64a1

But I can successfully create a team if I left things as they are:

![Screenshot 2024-07-03 at 10 45 42](https://github.com/JoinColony/colonyCDapp/assets/50642296/4e6ab5ea-66b6-461b-a737-838d9bfdcb7f)

Reply:
Ah okay, you might be correct on this one then! 👍 I didn't actually check the team got created after I finalised it.

Reply:
Tested and confirmed it does need the ternary.